samza-api/src/main/java/org/apache/samza/storage/kv/KeyValueStore.java

```diff
@@ -110,6 +110,18 @@ default void deleteAll(List<K> keys) {
    */
   KeyValueIterator<K, V> range(K from, K to);
 
+  /**
+   * Returns an iterable for a sorted range of entries specified by [{@code from}, {@code to}).
+   *
+   * @param from the key specifying the low endpoint (inclusive) of the keys in the returned range.
+   * @param to the key specifying the high endpoint (exclusive) of the keys in the returned range.
+   * @return an iterable for the specified key range.
+   * @throws NullPointerException if null is used for {@code from} or {@code to}.
+   */
+  default KeyValueIterable<K, V> iterate(K from, K to) {
```

Comment:
Iterable is a factory class which creates iterator, that's a java standard so I don't want to mess with it. I don't think it's a good idea at all to call it snapshot which implements a java iterable interface. Since it's not returning the snapshot, I don't want to call this function snapshot either. I am willing to take other names if iterate() is not clear enough.